contracts/vault/CurveSwapper.sol

```diff
@@ -179,4 +159,55 @@ abstract contract CurveSwapper {
             IERC20(_token).safeApprove(_pool, type(uint256).max);
         }
     }
+
+    /// @param _swapPools configs for each swap pool
+    function _addPools(SwapPoolParam[] memory _swapPools) internal {
+        for (uint256 i = 0; i < _swapPools.length; ++i) {
+            _addPool(_swapPools[i]);
+        }
+    }
+
+    function _addPool(SwapPoolParam memory _param) internal {
+        require(
+            address(swappers[_param.token].pool) == address(0),
+            "token already has a swap pool"
+        );
+        require(
+            ICurve(_param.pool).coins(uint256(uint128(_param.underlyingI))) ==
+                getUnderlying(),
+            "_underlyingI does not match underlying token"
+        );
+
+        uint256 tokenDecimals = IERC20Metadata(_param.token).decimals();
+        uint256 underlyingDecimals = IERC20Metadata(getUnderlying()).decimals();
+
+        // TODO check if _token and _underlyingIndex match the pool settings
+        swappers[_param.token] = Swapper(
+            ICurve(_param.pool),
+            uint8(tokenDecimals),
+            uint8(underlyingDecimals),
+            _param.tokenI,
```

Comment:
this is how curve works? we can store the index values permanently and expect them never to change?

Reply:
for a given curve pool, indexes are always the same, yes